internal/cloud/azure/application_identity.go

```diff
@@ -3,121 +3,149 @@ package azure
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/msi/armmsi"
 )
 
-type ApplicationClient interface {
-	Create(ctx context.Context, application msgraph.Application) (*msgraph.Application, int, error)
-	Get(ctx context.Context, id string, query odata.Query) (*msgraph.Application, int, error)
-	Update(ctx context.Context, application msgraph.Application) (int, error)
-	Delete(ctx context.Context, id string) (int, error)
+type ManagedIdentityClient interface {
+	CreateOrUpdate(ctx context.Context, resourceGroupName string, resourceName string, parameters armmsi.Identity, options *armmsi.UserAssignedIdentitiesClientCreateOrUpdateOptions) (armmsi.UserAssignedIdentitiesClientCreateOrUpdateResponse, error)
+	Get(ctx context.Context, resourceGroupName string, resourceName string, options *armmsi.UserAssignedIdentitiesClientGetOptions) (armmsi.UserAssignedIdentitiesClientGetResponse, error)
+	Delete(ctx context.Context, resourceGroupName string, resourceName string, options *armmsi.UserAssignedIdentitiesClientDeleteOptions) (armmsi.UserAssignedIdentitiesClientDeleteResponse, error)
 }
 
-func (c *AzureConfig) NewApplicationClient(ctx context.Context) (ApplicationClient, error) {
-	authorizer, authorizerErr := c.authConfig.NewAuthorizer(ctx, c.authConfig.Environment.MsGraph)
-	if authorizerErr != nil {
-		return nil, authorizerErr
-	}
-	appClient := msgraph.NewApplicationsClient(c.Provider.TenantID.Value)
-	appClient.BaseClient.Authorizer = authorizer
-	return appClient, nil
+type FederatedIdentityCredentialClient interface {
+	CreateOrUpdate(ctx context.Context, resourceGroupName string, resourceName string, federatedIdentityCredentialResourceName string, parameters armmsi.FederatedIdentityCredential, options *armmsi.FederatedIdentityCredentialsClientCreateOrUpdateOptions) (armmsi.FederatedIdentityCredentialsClientCreateOrUpdateResponse, error)
 }
 
-type ServicePrincipalsClient interface {
-	Create(ctx context.Context, servicePrincipal msgraph.ServicePrincipal) (*msgraph.ServicePrincipal, int, error)
-	Get(ctx context.Context, id string, query odata.Query) (*msgraph.ServicePrincipal, int, error)
-	Update(ctx context.Context, servicePrincipal msgraph.ServicePrincipal) (int, error)
-	Delete(ctx context.Context, id string) (int, error)
-
-	AddPassword(ctx context.Context, servicePrincipalId string, passwordCredential msgraph.PasswordCredential) (*msgraph.PasswordCredential, int, error)
-	RemovePassword(ctx context.Context, servicePrincipalId string, keyId string) (int, error)
+type ApplicationIdentityConfig struct {
+	// READ-ONLY; Fully qualified resource ID for the resource.
+	// /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}
+	ID                           string
+	Name                         string
+	Location                     string
+	ResourceGroupName            string
+	KubernetesNamspace           string
```

Comment:
Namespace is mispelled here.  I'd do a suggest change, but it will need to be updated in every reference. VSCode makes this easy w/ "Rename" and it will update all references (in case you didn't know).

Reply:
Great catch, thank you